include/opc/common/interface.h

```diff
@@ -18,9 +18,15 @@ namespace Common
   {
   protected:
     Interface(){}
+
+    // do _not_ allow destructors to throw exceptions as it
+    // creates a compiler error on OSX and may break shared_ptr
+    // cleanup and cause memory leaks
+    /*
     // c++11 destructors default to noexcept
     // allow implementations to do so.
     virtual ~Interface() noexcept (false) {}
+    */
 
```

Comment:
This was deliberately added in #205 because ~AddonsManagerImpl has a ``throw``, maybe it shouldn't.
Undoing this, destructor still needs to be declared virtual ``virtual ~Interface() {}`` ?